charts/vector/templates/keda.yaml

```
@@ -0,0 +1,34 @@
+{{- if or (eq .Values.role "Aggregator") (eq .Values.role "Stateless-Aggregator") -}}
+{{- if and .Values.keda.enabled (not .Values.autoscaling.enabled)}}
```

```suggestion
{{- if and .Values.keda.enabled (.Capabilities.APIVersions.Has "keda.sh/v1alpha1") (not .Values.autoscaling.enabled)}}
```

Does this work as well? I don't know if I've tried this with custom resources, but I assume it should work.

---

Good point, added